library/x509.c

```diff
@@ -753,6 +753,7 @@ int mbedtls_x509_dn_gets( char *buf, size_t size, const mbedtls_x509_name *dn )
 
     while( name != NULL )
     {
+        j = 0;
```

Comment:
Perhaps we could move this initialisation closer to the for loop where the variable is used? used?

Reply:
Perhaps, personally, I prefer to  have variables initialized at the first place that is possible, but I don't have strong feelings on that